### ISSUE TITLE
Uniformisation des fichiers

### ISSUE THREAD

**Issue by sbrodin** [2ec0b4/kaamelott-soundboard#136]
Je trouve que les noms des fichiers dans sounds sont un peu aléatoirement définis (underscore, tiret, majuscules, nom partiel ou complet de la réplique, etc.)
Est-ce qu'il ne faudrait pas uniformiser tout ça, auquel cas il faudrait se mettre d'accord sur une norme à adopter ?

**Comment by Aluriak** [comment]
Bonne idée. Ça peut se faire avec un petit code python. Ça m'occupera dans les jours prochains, tiens.

Je propose d'utiliser le format suivant : `CamelCasePourLesNomsDeFichiersOuSontEnlevesLAccentEtLApostrophe.mp3`

Pour effectuer la correction, je me propose de faire le code et le commit, une fois que le format sera déterminé par les contributeurs.

**Comment by sbrodin** [comment]
Je trouvais ça plus lisible en mettant des tirets ou des underscores. Et tout en minuscule et sans accent évidemment.
Après il faut s'entendre pour tous les remplacements d'apostrophes ("j'ai" ou "c'est" par exemple).
Et puis il faut faire quelque chose pour les fichiers qui peuvent avoir le même nom ("HÉRÉTIQUE ! AU BUCHER !" notamment).

**Comment by 2ec0b4** [comment]
J'avoue, je suis assez permissif sur le format des noms des fichiers.

Quand j'ai vu passer cette _issue_, j'ai pensé à @Aluriak et son dépôt https://github.com/Aluriak/kaamelott-soundboard-viz 😊

> Pour effectuer la correction, je me propose de faire le code et le commit, une fois que le format sera déterminé par les contributeurs.

🙏

> Je trouvais ça plus lisible en mettant des tirets ou des underscores. Et tout en minuscule et sans accent évidemment.

J'ai une préférence pour ce format également, en choisissant les tirets. D'autant que ça apparait dans l'URL. D'ailleurs, il y aura certainement une gestion de redirection à faire, si on veut faire ça bien 🤔

**Comment by 2ec0b4** [comment]
Du _Kebab case_ quoi

https://kaamelott-soundboard.2ec0b4.fr/#son/je_vois_pas_le_rapport_avec_bretagne

**Comment by Aluriak** [comment]
AJA Kebab case, que j'utilisais sans nommer pendant ma période lisp.

Donc on partirais sur du `kebab-case-ie-tiret-comme-separateur-et-pas-daccent-ni-apostrophe.mp3` ?

**Comment by Aluriak** [comment]
> Quand j'ai vu passer cette _issue_, j'ai pensé à @Aluriak et son dépôt https://github.com/Aluriak/kaamelott-soundboard-viz blush

Oui, va falloir que j'y rejette un œil. J'avais fait un truc qui réordonnait proprement le json, aussi.

**Comment by Aluriak** [comment]
> Après il faut s'entendre pour tous les remplacements d'apostrophes ("j'ai" ou "c'est" par exemple).

Je propose de les enlever, comme les accents.

> Et puis il faut faire quelque chose pour les fichiers qui peuvent avoir le même nom ("HÉRÉTIQUE ! AU BUCHER !" notamment).

C'est le plus chiant. Je dirais bien de suffixer par `-2`.

**Comment by 2ec0b4** [comment]
> AJA Kebab case, que j'utilisais sans nommer pendant ma période lisp.
> 
> Donc on partirais sur du `kebab-case-ie-tiret-comme-separateur-et-pas-daccent-ni-apostrophe.mp3` ?

👍

> Oui, va falloir que j'y rejette un œil. J'avais fait un truc qui réordonnait proprement le json, aussi.

Cf. https://github.com/2ec0b4/kaamelott-soundboard/pull/115

J'avais dit que j'allais mettre ça dans un crochet (_hook_ git) mais je ne l'ai pas fait 😅

> C'est le plus chiant. Je dirais bien de suffixer par `-2`.

Alors, je me pose la question suivante : est-ce qu'il faut mettre `-1` aussi du coup ? 🤔🙃

**Comment by Aluriak** [comment]
Nop, je pense pas. Inutile, d'autant que ça ne change pas leur ordre lexicographiquement. Bref, c'est la même chose fonctionnellement, et c'est IMHO plus clair.
